File: src/main/java/me/leodev/trollingfreedomreborn/commands/Freeze.java
package me.leodev.trollingfreedomreborn.commands;

import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.block.BlockBreakEvent;
import org.bukkit.event.block.BlockPlaceEvent;
import org.bukkit.event.player.PlayerMoveEvent;

import java.util.ArrayList;

public class Freeze implements Listener {

    public static ArrayList<String> frozen = new ArrayList();


    public void Freeze(Player p) {
        Player p2 = p.getPlayer();
        //p2.setFlying(true);
        frozen.add(p2.getName());
    }

    public void Unfreeze(Player p) {
        Player p2 = p.getPlayer();
        //p2.setFlying(false);
        frozen.remove(p2.getName());
    }

    @EventHandler
    public void onPlayerMove(PlayerMoveEvent e) {
        Player p = e.getPlayer();
        if (frozen.contains(p.getName())) {
            e.setTo(e.getFrom());
        }
    }

    @EventHandler
    public void onBlockBreak(BlockBreakEvent e) {
        Player p = e.getPlayer();
        if (frozen.contains(p.getName())) {
            e.setCancelled(true);
        }
    }

    @EventHandler
    public void onBlockPlace(BlockPlaceEvent e) {
        Player p = e.getPlayer();
        if (frozen.contains(p.getName())) {
            e.setCancelled(true);
        }
    }
}

File: src/main/java/com/leomadrassi/trollingfreedomreborn/trolls/movement/Freeze.kt
package com.leomadrassi.trollingfreedomreborn.trolls.movement

import org.bukkit.entity.Player
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.block.BlockBreakEvent
import org.bukkit.event.block.BlockPlaceEvent
import org.bukkit.event.player.PlayerMoveEvent

class Freeze : Listener {
    companion object {
        val frozen: MutableList<String> = ArrayList()
    }

    fun Freeze(p: Player) {
        frozen.add(p.name)
    }

    fun Unfreeze(p: Player) {
        frozen.remove(p.name)
    }

    @EventHandler
    fun onPlayerMove(e: PlayerMoveEvent) {
        val p = e.player
        if (frozen.contains(p.name)) {
            e.setTo(e.from)
        }
    }

    @EventHandler
    fun onBlockBreak(e: BlockBreakEvent) {
        val p = e.player
        if (frozen.contains(p.name)) {
            e.isCancelled = true
        }
    }

    @EventHandler
    fun onBlockPlace(e: BlockPlaceEvent) {
        val p = e.player
        if (frozen.contains(p.name)) {
            e.isCancelled = true
        }
    }
}
